### PR TITLE
feat(GCS+gRPC): implement `CreateDefaultObjectAcl()`

### LINKED ISSUE #4185: Implement storage::GrpcClient::CreateDefaultObjectAcl

### PR DESCRIPTION
**Pull request by coryan**
Fixes #4185

---
This change is [Reviewable](https://reviewable.io/reviews/googleapis/google-cloud-cpp/9465)

### CODE REVIEW

File: google/cloud/storage/internal/grpc_client.cc [hunk 1]
```diff
@@ -1015,6 +1021,28 @@ StatusOr<ObjectMetadata> GrpcClient::ModifyObjectAccessControl(
...
+  auto patch = PatchBucket(
+      PatchBucketRequest(
+          request.bucket_name(),
+          BucketMetadataPatchBuilder().SetDefaultAcl(*std::move(acl)))
+          .set_option(IfMetagenerationMatch(get->metageneration())));
+  // Retry on failed preconditions
+  if (patch.status().code() == StatusCode::kFailedPrecondition) {
+    return Status(
+        StatusCode::kUnavailable,
+        "retrying BucketAccessControl change due to conflict, bucket=" +
```
Comment (dbolduc) [inline google/cloud/storage/internal/grpc_client.cc:1040]: s/Bucket/Default/ ?

File: google/cloud/storage/internal/grpc_client.h [hunk 2]
```diff
@@ -198,6 +198,17 @@ class GrpcClient : public RawClient,
   StatusOr<ObjectMetadata> ModifyObjectAccessControl(
       GetObjectMetadataRequest const& request, ObjectAclUpdater const& updater);
 
+  using DefaultObjectAclUpdater =
+      std::function<StatusOr<std::vector<ObjectAccessControl>>(
+          std::vector<ObjectAccessControl> acl)>;
+
+  // REST has RPCs that change `BucketAccessControl` resources atomically. gRPC
```
Comment (dbolduc) [inline google/cloud/storage/internal/grpc_client.h:205]: s/Bucket/Default/ ?

**Review by dbolduc — APPROVED** [review]

File: google/cloud/storage/internal/grpc_client.cc (hunk 1 above)
Comment (coryan) [reply google/cloud/storage/internal/grpc_client.cc]: Fixed, thanks.

File: google/cloud/storage/internal/grpc_client.h (hunk 2 above)
Comment (coryan) [reply google/cloud/storage/internal/grpc_client.h]: Ditto

**Pull request merged**